src/object/object_primitive.c

```diff
@@ -10981,7 +10981,7 @@ mr_setval_string (DB_VALUE * dest, const DB_VALUE * src, bool copy)
     {
       error = db_value_domain_init (dest, DB_TYPE_VARCHAR, DB_DEFAULT_PRECISION, 0);
     }
-  else if (DB_IS_NULL (src) || (src_str = db_get_string (src)) == NULL)
+  else if (DB_IS_NULL (src) || ((src_str = db_get_string (src)) == NULL && src->data.ch.info.is_max_string == false))
```

Comment:
@lucianlazar2211 

Please examine `else` block. It seems it assumes `src_str` is not null, however, it will not set with your fix. 
Maybe it is better to have another `else if` block to handle `is_max_string` case.